### DISCUSSION TITLE
Which DB dialect to use in Springboot JPA ?

### DISCUSSION THREAD

**Discussion by PVerschueren107** [duckdb/duckdb#12675] [Q&A]
Hi
I am trying to setup a Springboot application using DuckDB.  One of the issues I encounter is that there doesnt seem to be a hibernate SQL Dialect for DuckDB out of the box.  I tried with MySQL and Oracle but these dont seem to work.  Anyone has any luck with other dialects or may even created their own ?
Kr
Peter

**Comment by szarnyasg** [comment]
DuckDB's dialect closely follows the PostgreSQL dialect, with [minor exceptions](https://duckdb.org/docs/sql/postgresql_compatibility).

**Reply by PVerschueren107** [reply]
Thank you for the reference to PostGreSQL.  Update the properties to reflect it.
However I think there might be a limitation in the JDBC driver implementation.
Once hibernate is up, it tries to get an SQL State resulting in Could not obtain connection metadata: java.sql.SQLFeatureNotSupportedException: getSQLStateType.  This error seems specific to the JDBC implementation.
Any ideas on that ?
Kr
Peter

**Reply by aby-kuruvilla-clear** [reply]
This specific function might not be implemented in the duckdb driver. You can try to clone the source code and maybe see if you could implement it or try passing a default value.
